src/components/Swap/Swap.tsx

```diff
@@ -531,19 +540,21 @@ export const Swap: React.FC<ISwap> = ({
                 }}
                 exchangeRate={{
                   val: swapRate,
-                  symbol: tokens[tokenToIndex].symbol
+                  symbol: tokens[toToken].symbol
                 }}
                 anchorTransaction={anchorTransaction}
                 handleCloseTransactionDetails={handleCloseTransactionDetails}
-                decimal={tokens[tokenToIndex].decimals}
+                decimal={tokens[toToken].decimals}
               />
-
               <Box className={classes.ableToHover}>
                 <ExchangeRate
-                  tokenFromSymbol={tokens[tokenFromIndex].symbol}
-                  tokenToSymbol={tokens[tokenToIndex].symbol}
+                  onClick={() => {
+                    swapTokens !== null ? setSwapTokens(!swapTokens) : setSwapTokens(true)
+                  }}
+                  tokenFromSymbol={tokens[fromToken].symbol}
+                  tokenToSymbol={tokens[toToken].symbol}
                   amount={swapRate}
```

Comment:
ok, I'm gonna once again explain how should look changes here:
1) only one useState - const [rateDirection, setRateDirection] = useState(true)
2) on click rate set rateDirection to its negation, and set it to true, when tokens from and to change
3) amount prop value in ExchangeRate should be swapRate when rateDirection === true and 1/swapRate, when false,
also swap token from and token to symbols
that should be enough changes, rest is unnecessary

Reply:
done